FTS-7 RestController "/api/pitches" GET endpoint

hulfyback opened the pull request:
There is an /api/pitches endpoint

And 4 tests:
- 2 for statuscodes
- 2 for response body

hulfyback committed:
feat(levelup): add pitches api

Create a PitchesRestController class for 'api/pitches' endpoint.
Create an interface containing a static final field returning a succesful response string.

hulfyback committed:
feat(levelup): add new fields in IPitchesRestController interface

Add new fields for:
- headername
- authorization status
- error body

hulfyback committed:
feat(levelup): repsonse header

Modify returntype of PitchesRestController @GetMapping method.

hulfyback committed:
refactor(levelup): outsource setHttpHeader method

Add new HttpHeaders field and setHttpHeader to make easier the test

hulfyback committed:
refacter(levelup): remove unused fields

Remove two static final String from IPitchesRestController interface

hulfyback committed:
refactor(levelup): delete method from PitchesRestController

Add isAuthorized boolean field
Set authorization
return new ResponseEntity()

hulfyback committed:
refactor(levelup): remove unused import

Remove @Mock annotation import

hulfyback committed:
test(levelup): test status is ok

There is a simply setAuthorized method in PitchesRestController
If isAuthorized == true -> response status is OK (code = 200)
Test if after set isAuthorized to true then it returns code 200 and gives back api body

hulfyback committed:
refactor(levelup): make static setAuthorized

PitchesRestController modify isAuthorized field to static and setAuthorized() to static

hulfyback committed:
test(levelup): test unauthorization

This test checks if status is unauthorized then returns error body

hulfyback committed:
test(levelup): test status codes

Test if response status is OK then returns statuscode 200
Test if response status is Unauthorized then returns statuscode 401

hulfyback committed:
refactor(levelup): rename RestController method

Rename getPpitchesApiError -> getPitchesApi

hulfyback committed:
style(levelup): rename controllerTest package

hulfyback committed:
refactor(levelup): refactor if statement

Remove 'true' from if(isAuthorized == true)

drexeverest reviewed (changes requested):
My comments from FTS6 could be applied here too

hulfyback committed:
fix(levelup): change set methods to header()

Instead of changing the value of 'Authorization' key in response header manually, MockMvc.header() method is used

hulfyback committed:
refactor(levelup): remove unused methods()

Remove IsAuthorized and httpHeaders fields
Remove setAuthorization() and httpHeaders.set() methods
Add real logic to check authorization

hulfyback committed:
chore(levelup): new Message class for constants

hulfyback committed:
chore(levelup): add constants to Message class

hulfyback committed:
chore(levelup): add new constants to Message

hulfyback committed:
chore(levelup): add business logic of RestController

hulfyback committed:
test(levelup): test status and response body

hulfyback committed:
refactor(levelup): outsource logic from RestController

hulfyback committed:
style(levelup): delete empty line

drexeverest commented on src/main/java/com/greenfoxacademy/levelup/utility/Util.java:
```diff
@@ -0,0 +1,19 @@
+public class Util {
+
+  public static ResponseEntity<String> isAuthorized(String authorization) {
+    {
```
{}

hulfyback committed:
refactor(levelup): move private method behind public

hulfyback committed:
style(levelup): remove whitespaces

add empty line at end of files

hulfyback committed:
style(levelup): add empty line

add an empty line at the end of the Message class

hulfyback committed:
Merge branch 'dev' into zsoltmolnar-FTS-7

resolve confilcts:
 Conflicts:
	src/main/java/com/greenfoxacademy/levelup/collection/Message.java
	src/main/java/com/greenfoxacademy/levelup/utility/Util.java

delete duplicated constants from Message
delete whitespaces from json format Strings in Message
change 'Authorization' String to constant in Controllers

hulfyback committed:
refactor(levelup): rename controller_test

rename controller_test -> controller in /main/test.. package

hulfyback committed:
merge(levelup): merge 'dev' to zsoltmolnar_FTS-7
